src/odemis/gui/comp/miccanvas.py

```diff
@@ -1165,6 +1172,227 @@ def update_drawing(self):
+
+    def set_data(self, data, unit_x=None, unit_y=None, range_x=None, range_y=None):
+        xs, ys = zip(*data)
+        self.set_1d_data(xs, ys, unit_x, unit_y, range_x, range_y)
+        self.refresh_plot()
+
+    def set_1d_data(self, xs, ys, unit_x=None, unit_y=None, range_x=None, range_y=None):
+        """
+        Note - this function will not plot any data until a call to refresh_plot is made
+        """
+
+        if len(xs) != len(ys):
+            msg = "X and Y list are of unequal length. X: %s, Y: %s, Xs: %s..."
+            raise ValueError(msg % (len(xs), len(ys), str(xs)[:30]))
+        self._data_buffer = (numpy.array(xs), numpy.array(ys))
+        self.unit_x = unit_x
+        self.unit_y = unit_y
+        
+        if range_x is None or range_y is None:
+            _, range_x, _, range_y = self._calc_data_characteristics(zip(xs, ys))
+
+        self.data_xrange = range_x
+        self.data_yrange = range_y
+        self.display_xrange = range_x
+        self.display_yrange = range_y
+        
+    def fit_view_to_content(self, recenter=None):
+        # note - need to do this via an event to the viewport. Otherwise it doesn't
+        # set the axes properly
+        evt = evtFitViewToContent()
+        wx.PostEvent(self, evt)
+
+    def get_hcontent_range(self):
+        """
+        To allow fitting to content, detect where the first leftmost non-zero value
+        occurs, and also detect where the first rightmost non-zero value occurs.
+        Returns the horizontal range that fits the data content
+        If no range is found, return none
+        """
+        if self._data_buffer is None:
+            return None
+
+        (xd , yd) = self._data_buffer
+        # If the empty portion on the edge is less than the threshold percent of
+        # the full data width, do not include it.
+        threshold = 0.05
+
+        for ileft in range(len(yd)):
+            if yd[ileft] > 0:
+                break
+        
+        for iright in range(len(yd) - 1, 0, -1):
+            if yd[iright] > 0:
+                break
+            
+        iright = len(yd) - iright
+            
+        # if ileft is > iright, then the data must be all 0.
+        if ileft < iright and (ileft / len(yd) < threshold \
+            or (1 - iright / len(yd)) < threshold):
+            return (xd[ileft], xd[iright])
+        else:
+            return None
+
+    def reset_ranges(self):
+        if self._data_buffer is None:
+            return
+
+        self.set_ranges(self.data_xrange, self.data_yrange)
+
+    def _resample_plot(self, xs, ys, threshold):
+        """
+        Re-sample the data if there are more points than the threshold
+        xs, ys: two 1-D arrays with data
+        threshold: the minimum number of points for a re-sampling to occur
+        """
+        # Re-sample data based on window size
+        if len(xs) > threshold:
+            rs_factor = len(xs) // self.ClientSize.x // 6
```

Comment:
I'd rather have the function just contain the content of this block (and pass `self.ClientSize.x * 6` as argument). The main advantage is that it'll not relay on  `self.ClientSize.x`, so it'll be generic, can be more easily reused. The other advantage is that it'd ensure that it's not possible to write `len(xs) // self.ClientSize.x // 6`, which is unreadable ;-) Instead, it'd be be `rs_factor = len(xs) // new_len`.

Reply:
Now I see that `threshold` is actually computed as `self.ClientSize.x * 10`. So you can just compute the rs_factor as ` len(xs) // threshold`, or  ` len(xs) // (threshold / 2)`